terraso_backend/apps/graphql/schema/story_maps.py

```diff
@@ -94,12 +131,170 @@ def mutate_and_get_payload(cls, root, info, **kwargs):
         user = info.context.user
         story_map = StoryMap.objects.get(pk=kwargs["id"])
 
-        if not user.has_perm(StoryMap.get_perm("delete"), obj=story_map):
+        if not rules.test_rule("allowed_to_delete_story_map", user, story_map):
             logger.info(
                 "Attempt to delete a StoryMap, but user lacks permission",
                 extra={"user_id": user.pk, "story_map_id": str(story_map.id)},
             )
             raise GraphQLNotAllowedException(
                 model_name=StoryMap.__name__, operation=MutationTypes.DELETE
             )
+
+        return super().mutate_and_get_payload(root, info, **kwargs)
+
+
+class StoryMapMembershipSaveMutation(BaseAuthenticatedMutation):
+    model_class = Membership
+    memberships = graphene.Field(graphene.List(CollaborationMembershipNode))
+
+    class Input:
+        user_role = graphene.String()
+        membership_status = graphene.String()
+        user_emails = graphene.List(graphene.String, required=True)
+        story_map_id = graphene.String(required=True)
+        story_map_slug = graphene.String(required=True)
+
+    @classmethod
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        user = info.context.user
+
+        if kwargs["user_role"] != ROLE_COLLABORATOR:
+            logger.info(
+                "Attempt to save Story Map Memberships, but user role is not collaborator",
+                extra={
+                    "user_role": kwargs["user_role"],
+                },
+            )
+            raise GraphQLNotAllowedException(
+                model_name=Membership.__name__, operation=MutationTypes.UPDATE
+            )
+
+        story_map_id = kwargs["story_map_id"]
+        story_map_slug = kwargs["story_map_slug"]
+
+        try:
+            story_map = StoryMap.objects.get(slug=story_map_slug, story_map_id=story_map_id)
+        except Exception as error:
+            logger.error(
+                "Attempt to save Story Map Memberships, but story map was not found",
+                extra={
+                    "story_map_id": story_map_id,
+                    "story_map_slug": story_map_slug,
+                    "error": error,
+                },
+            )
+            raise GraphQLNotFoundException(model_name=StoryMap.__name__)
```

Comment:
suggestion: I tried adding a method to the parent class `get_or_throw` that I think you could use here.

Reply:
Great, I will check that one